float32[] rhs
float32[] ths
int16 imageSizeX
int16 imageSizeY
int16 nodePosX
int16 nodePosY
float32[] exitDirs

